docs/building-open-mpi/required-support-libraries.rst

```diff
@@ -0,0 +1,128 @@
+  obscure corner cases.  If this has happened to you, continue reading
+  this section.
+* If you have no problems with Open MPI and these four support
+  libraries, then you can safely skip reading this section.
+
+Bundled versions
+^^^^^^^^^^^^^^^^
+
+These libraries are bundled in Open MPI because they are *required*
+for correct Open MPI functionality (with the exception of PRRTE, which
+can be omitted if another PMIx-compatible run-time system is
+available, such as SLURM), and are not universally available by
+default on many systems.  For example, macOS does not include
+any of these libraries.
+
+Rather than make end-users separately download and install these
+required libraries, Open MPI bundles them in its distribution tarball,
+and will automatically use the bundled versions if compatible versions
+cannot be found installed on the build system.
+
+System-provided versions
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+That being said, many systems *do* provide installed versions of these
+libraries.  For that reason, Open MPI "prefers" to use system-provided
+versions of these libraries |mdash| if it can find them |mdash|
+instead of the bundled versions.
+
+Meaning: if ``configure`` finds suitable system-provided support
+libraries, it will use them by default, and ignore the bundled
+versions.
+
+Dependencies between support libraries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+These support libraries do have dependencies upon each other:
+
+* PMIx uses Hwloc and Libevent.
+* PRRTE uses PMIx, Hwloc, and Libevent.
+
+There are two important consequences to these dependencies:
+
+#. If the system-provided PMIx and PRRTE are used, they *must* use the
+   same Hwloc and Libevent installations.
+
+   .. danger:: If they do not, Open MPI will behave unpredictably.
+
+#. If ``configure`` chooses to use a system-provided PMIx or a
+   system-provided PRRTE, then ``configure`` will abort if it is
+   unable to find system-provided headers and libraries for *all four
+   support libraries*.
+
+   Put differently: ``configure`` will ignore *all* bundled support
+   libraries if a suitable external PMIx installation or suitable
+   external PRRTE installation is found.
+
+
+Which to use?
+^^^^^^^^^^^^^
+
+From a performance perspective, there is no significant reason to
+choose the bundled vs. system-installed installations.  Specifically:
+all will likely give the same performance.  They're really only
+bundled for end-user installation convenience.
+
+That being said, there definitely are obscure technical corner cases
+and philosophical reasons to force the choice of one or the other.  As
+such, Open MPI provides ``configure`` command line options that can be
+used to specify exact behavior in searching for hwloc and/or Libevent.
+
+.. important:: If you are a downstream packager such as a Linux
+   distribution, Homebrew or MacPorts packager, or other OS/HPC
+   packager, you should use system-provided versions of these packages
```

Comment:
Rather than system-provided (remember, these are the system providers), say "we suggest that the Open MPI package should not include libhwloc, libevent, pmix, or prrte, but should instead depend on independently built versions of those packages".